streamer/src/nonblocking/quic.rs

```diff
@@ -924,6 +925,26 @@ impl ConnectionTable {
     }
 }
 
+pub trait SolanaQuicCryptoConfig<Side: rustls::ConfigSide> {
+    fn with_solana_quic_crypto_config(self) -> rustls::ConfigBuilder<Side, WantsVerifier>;
+}
+
+impl<Side> SolanaQuicCryptoConfig<Side> for rustls::ConfigBuilder<Side, WantsCipherSuites>
+where
+    Side: rustls::ConfigSide,
+{
+    fn with_solana_quic_crypto_config(self) -> rustls::ConfigBuilder<Side, WantsVerifier> {
+        self.with_cipher_suites(&[
+            TLS13_AES_128_GCM_SHA256,
+            TLS13_CHACHA20_POLY1305_SHA256,
+            TLS13_AES_256_GCM_SHA384,
+        ])
+        .with_kx_groups(&ALL_KX_GROUPS)
+        .with_protocol_versions(&[&TLS13])
```

Comment:
The default supports both TLS1.2 and TLS1.3, do we lose anything with this change?

Reply:
just a bunch of obsolete/insecure/disused cipher suites, hash algos and protocol features.  iirc the main impetus behind 1.3 was to make tls less difficult to use safely